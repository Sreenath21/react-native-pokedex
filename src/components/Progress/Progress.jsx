import React from 'react';
import {View, Text} from 'react-native';

import styles from './Progress.style';

const Progress = ({value}) => {
  return (
    <View style={[styles.container]}>
      <View style={[styles.progress, {width: `${value}%`}]} />
      <Text style={styles.value}>{value}</Text>
    </View>
  );
};

export default Progress;
