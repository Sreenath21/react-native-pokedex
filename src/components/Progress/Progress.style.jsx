import {StyleSheet} from 'react-native';
import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    height: 12,
    backgroundColor: colors.progressEmpty,
    overflow: 'hidden',
    flex: 10,
  },
  progress: {
    height: 12,
    flex: 1,
    width: '100%',
    backgroundColor: colors.primary,
    position: 'absolute',
    top: 0,
    left: 0,
  },
  value: {
    color: '#fff',
    fontSize: 10,
    marginLeft: 5,
    fontWeight: 500,
  },
});
