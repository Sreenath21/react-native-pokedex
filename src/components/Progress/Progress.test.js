import React from 'react';
import Progress from './Progress';
import renderer from 'react-test-renderer';

it('Progress component renders correctly', () => {
  const tree = renderer.create(<Progress value={80} />);

  expect(tree).toMatchSnapshot();
});
