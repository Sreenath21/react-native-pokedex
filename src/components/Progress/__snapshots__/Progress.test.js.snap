// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress component renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#9fc6c7",
        "flex": 10,
        "height": 12,
        "overflow": "hidden",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#391363",
          "flex": 1,
          "height": 12,
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
        {
          "width": "80%",
        },
      ]
    }
  />
  <Text
    style={
      {
        "color": "#fff",
        "fontSize": 10,
        "fontWeight": 500,
        "marginLeft": 5,
      }
    }
  >
    80
  </Text>
</View>
`;
